---
title: "Fig_5G_v2"
output: github_document
date: "2024-02-06"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Loading in required packages
```{r}
library(conflicted)
library(tidyverse)
library(ggbio)
library(EnsDb.Hsapiens.v86)
library(patchwork)
conflicts_prefer(ggbio::autoplot)
conflicts_prefer(ggplot2::xlim)
conflicts_prefer(ggplot2::geom_rect)
conflicts_prefer(ggbio::geom_segment)
```


## Loading in data
```{r}
#Methylation frequency for sample of interest separated by haplotype for NRG1 region:
Methyl_freq_NRG1 <- read.delim("https://www.bcgsc.ca/downloads/nanopore_pog/ecDNA/POG816_NRG1_haplotype_methylation_source_table.txt", header = T, stringsAsFactors = F)

#CN calls from Ploidetect for sample of interest for NRG1 region:
CN_NRG1 <- read.delim("https://www.bcgsc.ca/downloads/nanopore_pog/ecDNA/POG816_NRG1_CN_ploidetect_source_table.txt", header = T, stringsAsFactors = F)
```

## Composing plot

```{r}
#Define colours to be used in plot:
colours <- c("#8A7EC7", "#7C3A88")

#Gene annotation for NRG1:
gr <- GRanges(seqnames = 8, IRanges(31637245, 32776046), strand = "*")
ensdb <- EnsDb.Hsapiens.v86

r <- autoplot(ensdb, GRangesFilter(gr), names.expr = "NRG1", xlab = "NRG1", label = FALSE) + theme_bw(base_size=11) + theme(legend.title = element_blank(), legend.position = "none", axis.title.x = element_text(colour = "black", size = 11), axis.line.y = element_blank(), 
axis.title.y = element_text(colour = "black", size = 11, face = "bold"), axis.line.x = element_blank(), panel.grid.major.x = element_blank(), panel.grid.minor.x = element_blank(), axis.text.x = element_blank(), panel.grid.minor.y = element_blank(), axis.ticks.x = element_blank(), plot.background = element_rect(colour = NA), panel.border = element_blank())

#Methylation frequency for NRG1 separated by haplotype (ID column):
q <- ggplot(Methyl_freq_NRG1, aes(start, MethylFreq, fill = ID)) + 
  xlim(31637245, 32776046) + 
  geom_smooth(method="loess", aes(colour = ID), se=F, size=1.5, span=0.5) + 
  scale_colour_manual(values = colours) + 
  annotate("text", label = "DMR1", x = 31639573, y = 1.04, size = 2.5) + 
  annotate("text", label = "DMR2", x = 32219817, y = 1.04, size = 2.5) + 
  scale_fill_manual(values = colours) + 
  labs(x = "Coordinate on chromosome 8", y = "Methylation frequency", fill = "Haplotype") + 
  theme_bw(base_size=11) + 
  theme(legend.title = element_blank(), 
        legend.position = "none", 
        axis.title.x = element_text(colour = "black", size = 11, face = "bold"), 
        axis.line.y = element_line(), 
        axis.title.y = element_text(colour = "black", size = 11, face = "bold"), 
        axis.line.x = element_line(), 
        panel.grid.major.x = element_blank(), 
        panel.grid.minor.x = element_blank(), 
        panel.grid.major.y = element_blank(), 
        panel.grid.minor.y = element_blank(), 
        axis.ticks.x = element_line(), 
        plot.background = element_rect(colour = NA), 
        panel.border = element_blank())

#Adding in two promoter DMR annotations:
s <- q + 
  geom_rect(xmin = 31638222, xmax = 31639221, ymin = 0, ymax = 1, colour = "yellow") + 
  geom_rect(xmin = 32220042, xmax = 32221041, ymin = 0, ymax = 1, colour = "yellow")

#Faceting by haplotype:
t <- s + facet_grid(rows = vars(ID))

#Adding in CN track:
u <- ggplot(CN_NRG1, aes(pos, CN)) + scale_x_continuous(limits = c(31637245, 32776046)) + ylim(0,10) + geom_segment(aes(x = pos, y = CN, xend = end, yend = CN), data = CN_NRG1) + theme_bw(base_size=11) + theme(legend.title = element_text(), axis.title.x = element_blank(), axis.line.y = element_line(), axis.title.y = element_text(colour = "black", size = 11, face = "bold"), axis.line.x = element_blank(),panel.grid.major.x = element_blank(), panel.grid.minor.x = element_blank(), axis.text.x = element_blank(), panel.grid.major.y = element_blank(), panel.grid.minor.y = element_blank(), axis.ticks.x = element_blank(), plot.background = element_rect(colour = NA), panel.border = element_blank())

#Combining plots together into one plot
patch <- (r@ggplot / u / t )
```
## Printing plot
```{r}
patch

```
