---
title: "ASE_karyogram.Rmd"
output: html_document
---
## ASE karyogram 

This R Markdown is used to generate the karyogram figure for POG184. The figure uses ASE results from IMPALA, allelic methylation data from NanoMethPhase and copy number variant data from Ploidetect. The following Rmd is copied/modified from the `karyogramFigure.R` script in the IMPALA pipeline.


```{r setup}
suppressMessages(library(dplyr))
suppressMessages(library(reshape2)) 
suppressMessages(library(ggplot2))
suppressMessages(library(tidyr))
suppressMessages(library(cowplot))
```

This import the required reference files an intermediate files generated from IMPALA
```{r import-data}
SAMPLE <- # NEED TO ADD 

# Reference file
chromSize <- read.delim("ref/hg38_genome_length.tsv", header = F)
centPos <-  read.delim("ref/hg38_centromere_positions.bed", header = F)
genes <- read.delim("ref/hg38_gene_annotation.bed", header = F)

# Import data
cna <- read.delim("/projects/lculibrk_prj/pog_cron/hg38/Ploidetect-pipeline/ploidetect_out_1_3_0/POG184/P00537_P00499/cna_condensed.txt", header = T)
dmr <- read.delim("/projects/nanopore_pog/analysis/Methylation/POG184-FF-1_F112341_F94404/Allelic_DMA_DSS-2-46/POG184-FF-1_F112341_F94404_callDMR_Filtered_diff.methyl0.15.txt.gz", header = T)
ase <- readRDS("/projects/glchang_prj/finalPOGdata/allASEdata.RDS") %>%
  dplyr::filter(sample == SAMPLE)


```

Obtain chromosome position information from reference file
```{r}
# subset to the main chromosomes
chromSize <- chromSize[chromSize$V1 %in% c(paste0("chr", 1:22), "chrX"),]

# Rename columns to chromosome and size
colnames(chromSize) <- c("chr","size")

# Reorder levels for plotting
chromSize$chr <- factor(chromSize$chr,levels=c(paste0("chr", 1:22), "chrX"))
chromSize <- chromSize[chromSize$chr %in% c(paste0("chr", 1:22), "chrX"),]

# Divide by 1Mb to clean up axis
chromSize$size <- chromSize$size/1000000

# centromere mapping
colnames(centPos) <- c("chr", "start", "end")
centPos$chr <- factor(centPos$chr,levels=c(paste0("chr", 1:22), "chrX"))
centPos$centre <- centPos$start + ((centPos$end - centPos$start)/2)
centPos$centre <- centPos$centre/1000000
```

# Process CNV data 
```{r}
cna$chr <- paste0("chr", cna$chr)

# rearrange to make a bed file
cna_bed <- cna[,c("chr", "pos", "end", "CN", "zygosity", "A", "B")]

# categorize copy number
cna_bed <- cna_bed %>%
  mutate(CN.Status =
           case_when(
             zygosity == "HOM" ~ "LOH",
             A > B ~ "imbalance",
             TRUE ~ "balance"
           ))

# Divide by 1Mb for axis
cna_bed$end <- cna_bed$end/1000000
cna_bed$pos <- cna_bed$pos/1000000

# Change to factor and reorder levels
cna_bed$chr <- factor(cna_bed$chr,levels=c(paste0("chr", 1:22), "chrX"))

cnaLOH <- cna_bed %>% filter(CN.Status == "LOH")
cnaGAIN <- cna_bed %>% filter(CN.Status == "imbalance")
```

# Process allelic methylation data
```{r}
# Divide by 1Mb for axis
dmr$start <- dmr$start/1000000
dmr$end <- dmr$end/1000000
dmr$middle <- (dmr$start + dmr$end) / 2

# Change to factor and reorder levels
dmr$chr <- factor(dmr$chr, levels=c(paste0("chr", 1:22), "chrX"))

# count in 1Mb bins
dmrCount <- data.frame(table(as.factor(paste0(dmr$chr, ":", as.integer(dmr$middle)))))

# split the chromosome name and bin position
dmrPlot <- separate(dmrCount, col = Var1, into = c("chr", "pos"), sep = ":", remove = T)

# scale to fit the plot - i.e. make the maximum width 0.65
maxDMR <- max(dmrPlot$Freq)
dmrPlot$percMax <- dmrPlot$Freq/maxDMR
dmrPlot$percMax <- dmrPlot$percMax * 0.65

# Change to factor and reorder levels
dmrPlot$chr <- factor(dmrPlot$chr, levels=c(paste0("chr", 1:22), "chrX"))
dmrPlot$pos <- as.numeric(dmrPlot$pos)
```

# Process allelic expression data
```{r}
# filter for ASE genes
ase <- ase[ase$aseResults == "ASE",]

# get gene positions
ase$chr <- genes$V1[match(ase$gene, genes$V4)]
ase$start <- genes$V2[match(ase$gene, genes$V4)]
ase$end <- genes$V3[match(ase$gene, genes$V4)]

# get the middle of the gene for plotting
ase$middle <- (ase$start + ase$end)/2

# get the data frame ready for plotting
ase <- ase[,c("chr", "start", "end","middle")]
ase <- ase[complete.cases(ase),]

# Divide by 1Mb for axis
ase$middle <- ase$middle/1000000

# count in 1Mb bins
aseCount <- data.frame(table(as.factor(paste0(ase$chr, ":", as.integer(ase$middle)))))

# split the chromosome name and bin position
asePlot <- separate(aseCount[aseCount$Var1 != "NA:NA",], col = Var1, into = c("chr", "pos"), sep = ":", remove = T)

# scale to fit the plot - i.e. make the maximum width 0.65
maxASE <- max(asePlot$Freq)
asePlot$percMax <- asePlot$Freq/maxASE
asePlot$percMax <- asePlot$percMax * 0.65

# Change to factor and reorder levels
asePlot$chr <- factor(asePlot$chr, levels=c(paste0("chr", 1:22), "chrX"))
asePlot$pos <- as.numeric(asePlot$pos)
```

# Plot Figure
```{r}
adL <- data.frame(xmin = c(9.7, 9.7, 9.7, 10.3,9.7,10.3,7.1,7.1), xmax = c(10.35, 10.35,9.75,10.35,9.75,10.35,7.26,7.26), ymin = c(240,210,238,238,208,208,235,205), ymax = c(243,213,243,243,213,213,245,215),
                    fill = c("ase","dmr","ase","ase","dmr","dmr", "gain", "loh"))
  
adW <- data.frame(x = c(11.5, 11.2,8.25,7.6,10,10), y = c(240,210,240,210,250,220),
                label = c("ASE Gene Density","DMR Density", "Imbalanced CNV", "LOH", as.character(c(maxASE,maxDMR))))

# plot
# chromosomes 1 - 12
p1 <- ggplot() +
# chromosome bars
geom_segment(data = chromSize %>% filter(chr %in% paste0("chr", 1:12)), aes(x = chr, xend = chr, y = 0, yend = size), 
             lineend = "round", color = "lightgrey", size = 5) +
# LOH
geom_rect(data = cnaLOH %>% filter(chr %in% paste0("chr", 1:12)), 
          aes(xmin = as.integer(chr) - 0.08, xmax = as.integer(chr) + 0.08, ymin = pos, ymax = end),
          fill="#94d2bd",size = 0.2) +
# Imbalanced CNV
geom_rect(data = cnaGAIN %>% filter(chr %in% paste0("chr", 1:12)), 
          aes(xmin = as.integer(chr) - 0.08, xmax = as.integer(chr) + 0.08, ymin = pos, ymax = end),
          fill="#ee9b00",size = 0.2) +
# ASE genes
geom_rect(data = asePlot %>% filter(chr %in% paste0("chr", 1:12)), 
          aes(xmin = as.integer(chr) + 0.1, xmax = (as.integer(chr) + 0.1 + percMax), ymin = pos, ymax = pos+1),
          fill = "#005f73", size = 0.25) +
# DMRs
geom_rect(data = dmrPlot %>% filter(chr %in% paste0("chr", 1:12)), 
          aes(xmin = (as.integer(chr) - 0.1 - percMax), xmax = as.integer(chr) - 0.1, ymin = pos, ymax = pos+1),
          fill = "#ae2012", size = 0.25) +
# centromeres
geom_point(data = centPos %>% filter(chr %in% paste0("chr", 1:12)), aes(x = chr, y = centre), 
           size = 5, colour = "gray40") +
# legend bars
geom_rect(data = adL, 
          aes(xmin = xmin, xmax = xmax, ymin = ymin, ymax = ymax, fill = fill),
          size = 0.25) +
# legend text
geom_text(data = adW, 
          aes(x = x, y = y, label = label))+
scale_fill_manual(values = c("#005f73","#ae2012","#ee9b00","#94d2bd")) +
ylim(0, 250) +
theme_classic() +
theme(text = element_text(size=15),axis.line=element_blank(),
      axis.ticks.x=element_blank(),
      legend.position = "none")+
labs(x=NULL,y="Chromosome Size (Mb)")

# chromosomes 13 - 22 + X

# very annoying but you have to filter all the dataframes or else the factor levels won't match the integer value
chromSizeFilt <- chromSize %>% filter(chr %in% c(paste0("chr", 13:22), "chrX"))
chromSizeFilt$chr <- factor(chromSizeFilt$chr,levels=c(paste0("chr", 13:22), "chrX"))
centPosFilt <- centPos %>% filter(chr %in% c(paste0("chr", 13:22), "chrX"))
centPosFilt$chr <- factor(centPosFilt$chr,levels=c(paste0("chr", 13:22), "chrX"))
cnaLOHFilt <- cnaLOH %>% filter(chr %in% c(paste0("chr", 13:22), "chrX"))
cnaLOHFilt$chr <- factor(cnaLOHFilt$chr,levels=c(paste0("chr", 13:22), "chrX"))
cnaGAINFilt <- cnaGAIN %>% filter(chr %in% c(paste0("chr", 13:22), "chrX"))
cnaGAINFilt$chr <- factor(cnaGAINFilt$chr,levels=c(paste0("chr", 13:22), "chrX"))
asePlotFilt <- asePlot %>% filter(chr %in% c(paste0("chr", 13:22), "chrX"))
asePlotFilt$chr <- factor(asePlotFilt$chr,levels=c(paste0("chr", 13:22), "chrX"))
dmrPlotFilt <- dmrPlot %>% filter(chr %in% c(paste0("chr", 13:22), "chrX"))
dmrPlotFilt$chr <- factor(dmrPlotFilt$chr,levels=c(paste0("chr", 13:22), "chrX"))

# chromosomes 13-22+X
p2 <- ggplot() +
# chromosome bars
geom_segment(data = chromSizeFilt, aes(x = chr, xend = chr, y = 0, yend = size), 
             lineend = "round", color = "lightgrey", size = 5) +
# LOH
geom_rect(data = cnaLOHFilt, 
          aes(xmin = as.integer(chr) - 0.08, xmax = as.integer(chr) + 0.08, ymin = pos, ymax = end),
          fill="#94d2bd",size = 0.2) +
# Imbalanced CNV
geom_rect(data = cnaGAINFilt, 
          aes(xmin = as.integer(chr) - 0.08, xmax = as.integer(chr) + 0.08, ymin = pos, ymax = end),
          fill="#ee9b00",size = 0.2) +
# ASE genes
geom_rect(data = asePlotFilt, 
          aes(xmin = as.integer(chr) + 0.1, xmax = (as.integer(chr) + 0.1 + percMax), ymin = pos, ymax = pos+1),
          fill = "#005f73", size = 0.25) +
# DMRs
geom_rect(data = dmrPlotFilt, 
          aes(xmin = (as.integer(chr) - 0.1 - percMax), xmax = as.integer(chr) - 0.1, ymin = pos, ymax = pos+1),
          fill = "#ae2012", size = 0.25) +
# centromeres
geom_point(data = centPosFilt, aes(x = chr, y = centre), 
           size = 5, colour = "gray40") +
ylim(0, 250) +
theme_classic() +
theme(text = element_text(size=15),axis.line=element_blank(),
      axis.ticks.x=element_blank(),
      legend.position = "none")+
labs(x=NULL,y=NULL)

plot <- plot_grid(p1, p2, align = "v", axis = "l", nrow = 2)

plot
```