{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-03-05T01:01:49.989560Z",
     "iopub.status.busy": "2024-03-05T01:01:49.989409Z",
     "iopub.status.idle": "2024-03-05T01:01:50.662356Z",
     "shell.execute_reply": "2024-03-05T01:01:50.661800Z"
    }
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import logging\n",
    "import numpy as np\n",
    "from matplotlib.pylab import plt\n",
    "from statsmodels.nonparametric.smoothers_lowess import lowess\n",
    "%matplotlib inline\n",
    "\n",
    "params = {\"font.family\":\"Arial\",\n",
    "\"legend.fontsize\": 10, \n",
    "        \"axes.labelsize\": 12,\n",
    "        \"axes.titlesize\": 12,\n",
    "        \"xtick.labelsize\": 12,\n",
    "        \"ytick.labelsize\": 12,\n",
    "        \"svg.fonttype\": 'none'}\n",
    "# plt.rcParams['svg.fonttype'] = 'none'\n",
    "plt.rcParams.update(params)\n",
    "\n",
    "logger = logging.getLogger(__name__)\n",
    "logging.basicConfig(level=logging.INFO)\n",
    "\n",
    "### COLORS #### \n",
    "TUMOUR_COLOR_HP1 = \"#7C3A88\"\n",
    "TUMOUR_COLOR_HP2 = \"#8A7EC7\"\n",
    "NORMAL_COLOR_HP1= \"#69BDC3\"\n",
    "NORMAL_COLOR_HP2 = \"#0B8892\"\n",
    "\n",
    "### CASES WITH PROMOTER METHYLATION\n",
    "BRCA1_METHYLATED_CASES_BRC_OVA = [\"POG804\", \"POG507\", \"POG425\"]\n",
    "BRCA1_METHYLATED_CASES_NON_BRC_OVA = [\"POG277\", \"POG1041\", \"POG650\"]\n",
    "BRCA1_LOH_CASES = [\"POG804\", \"POG507\", \"POG425\", \"POG1041\", \"POG650\"]\n",
    "\n",
    "RAD51C_METHYLATED_CASES_BRC_OVA = [\"POG356\", \"POG894\", \"POG846\"]\n",
    "RAD51C_METHYLATED_CASES_NON_BRC_OVA = [\"POG785\", \"POG266\", \"POG044\"]\n",
    "RAD51C_LOH_CASES = [\"POG356\", \"POG894\", \"POG846\"]\n",
    "MLH1_METHYLATED_CASES = [\"POG986\", \"POG041\"]\n",
    "MLH1_LOH_CASE = [\"POG986\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-03-05T01:01:50.665288Z",
     "iopub.status.busy": "2024-03-05T01:01:50.664883Z",
     "iopub.status.idle": "2024-03-05T01:01:50.685101Z",
     "shell.execute_reply": "2024-03-05T01:01:50.684637Z"
    }
   },
   "outputs": [],
   "source": [
    "################\n",
    "#### BRCA1 #####\n",
    "################\n",
    "\n",
    "def generate_BRCA1_haplotype_plots(BRCA1_files, BRCA1_methylated_cases_BRC_OVA, BRCA1_methylated_cases_non_BRC_OVA, BRCA1_loh_cases):\n",
    "    \n",
    "    tumour_types = [\"BRC_OVA\", \"non_BRC_OVA\"]\n",
    "\n",
    "    for tumour_type in tumour_types:\n",
    "        df_blood = pd.DataFrame()\n",
    "        df_tumour = pd.DataFrame()\n",
    "        \n",
    "        if tumour_type == \"BRC_OVA\":\n",
    "            ### Figure 5c\n",
    "            case_list = BRCA1_methylated_cases_BRC_OVA\n",
    "            \n",
    "        else:\n",
    "            ### Ex Fig 5b\n",
    "            case_list = BRCA1_methylated_cases_non_BRC_OVA\n",
    "        \n",
    "        files_BRCA1 = BRCA1_files[tumour_type]\n",
    "        logger.info(\"generating BRCA1 haplotype plots for {}\".format(tumour_type))\n",
    "        \n",
    "        if len(files_BRCA1[0]) >0:\n",
    "            df_blood = pd.read_csv(files_BRCA1[0], sep = \"\\t\", index_col = [\"gene\",\"start\", \"end\"])\n",
    "        if len(files_BRCA1[1])>0:\n",
    "            df_tumour = pd.read_csv(files_BRCA1[1], sep = \"\\t\", index_col = [\"gene\",\"start\", \"end\"])\n",
    "\n",
    "    \n",
    "        fig_BRCA1, ax_BRCA1 = plt.subplots(ncols=1, nrows=3, sharex = True, figsize = (8.5, 4.2), dpi = 600)\n",
    "    \n",
    "        for patient_index, patient in enumerate(case_list):\n",
    "            \n",
    "            LOH = \"\"\n",
    "            \n",
    "            hp1_column = f\"{patient}_HP1\"\n",
    "            hp2_column = f\"{patient}_HP2\"\n",
    "            \n",
    "            if patient in BRCA1_loh_cases:\n",
    "                LOH = \"yes\"\n",
    "            \n",
    "            if not df_tumour.empty:\n",
    "                hp1 = df_tumour[hp1_column]\n",
    "                hp2 = df_tumour[hp2_column]\n",
    "                xpos = [value[1] for value in df_tumour.index]\n",
    "               \n",
    "                ### LOESS SMOOTHING TUMOUR\n",
    "                filtered_hp1 = lowess(hp1, xpos, is_sorted=True, frac=0.25, it=0)\n",
    "                x_hp1 = filtered_hp1[:, 0]\n",
    "                y_hp1 = filtered_hp1[:, 1]\n",
    "        \n",
    "                filtered_hp2 = lowess(hp2, xpos, is_sorted=True, frac=0.25, it=0)\n",
    "                x_hp2 = filtered_hp2[:, 0]\n",
    "                y_hp2 = filtered_hp2[:, 1]\n",
    "                \n",
    "            if not df_blood.empty:\n",
    "                if hp1_column in df_blood.columns:\n",
    "                    hp1_blood = df_blood[hp1_column]\n",
    "                    hp2_blood = df_blood[hp2_column]\n",
    "                    xpos_blood = [value[1] for value in df_blood.index]\n",
    "            \n",
    "                    ### LOESS SMOOTHING BLOOD\n",
    "                    filtered_hp1_blood = lowess(hp1_blood, xpos_blood, is_sorted=True, frac=0.25, it=0)\n",
    "                    x_hp1_blood = filtered_hp1_blood[:, 0]\n",
    "                    y_hp1_blood = filtered_hp1_blood[:, 1]\n",
    "                \n",
    "                    filtered_hp2_blood = lowess(hp2_blood, xpos_blood, is_sorted=True, frac=0.25, it=0)\n",
    "                    x_hp2_blood = filtered_hp2_blood[:, 0]\n",
    "                    y_hp2_blood = filtered_hp2_blood[:, 1]\n",
    "\n",
    "            if not df_tumour.empty:\n",
    "                if LOH == \"yes\":\n",
    "                    if patient != \"POG507\":\n",
    "                        ax_BRCA1[patient_index].plot(x_hp1, y_hp1, \"-\", color = TUMOUR_COLOR_HP1, markerfacecolor=TUMOUR_COLOR_HP1, mec = TUMOUR_COLOR_HP1, marker=\"o\", markersize = 4, linewidth = 0.8, label = \"HP1_tumour\")\n",
    "                        ax_BRCA1[patient_index].plot(x_hp2, y_hp2, \"-\", color = TUMOUR_COLOR_HP2, markerfacecolor=\"none\", mec = TUMOUR_COLOR_HP2, mew = 0.5, marker=\"o\", markersize = 4.5, linewidth = 0.8, label = \"HP2_tumour\")\n",
    "    \n",
    "                    ### flipping color because the haplotype was aribitary set for the cases\n",
    "                    else:\n",
    "                        ax_BRCA1[patient_index].plot(x_hp2, y_hp2, \"-\", color = TUMOUR_COLOR_HP1, markerfacecolor=TUMOUR_COLOR_HP1, mec = TUMOUR_COLOR_HP1, marker=\"o\", markersize = 4, linewidth = 0.8, label = \"HP1_tumour\")\n",
    "                        ax_BRCA1[patient_index].plot(x_hp1, y_hp1, \"-\", color = TUMOUR_COLOR_HP2, markerfacecolor=\"none\", mec = TUMOUR_COLOR_HP2, mew = 0.5, marker=\"o\", markersize = 4.5, linewidth = 0.8, label = \"HP2_tumour\")\n",
    "                else:\n",
    "                    ax_BRCA1[patient_index].plot(x_hp1, y_hp1, \"-\", color = TUMOUR_COLOR_HP1, markerfacecolor=TUMOUR_COLOR_HP1, mec = TUMOUR_COLOR_HP1, marker=\"o\", markersize = 4, linewidth = 0.8, label = \"HP1_tumour\")\n",
    "                    ax_BRCA1[patient_index].plot(x_hp2, y_hp2, \"-\", color = TUMOUR_COLOR_HP2, markerfacecolor=TUMOUR_COLOR_HP2, mec = TUMOUR_COLOR_HP2, marker=\"o\", markersize = 4, linewidth = 0.8, label = \"HP2_tumour\")\n",
    "                    \n",
    "            if not df_blood.empty:\n",
    "                    ax_BRCA1[patient_index].plot(x_hp1_blood, y_hp1_blood, \"-\", color = NORMAL_COLOR_HP1, markerfacecolor=NORMAL_COLOR_HP1, mec = NORMAL_COLOR_HP1, marker=\"^\", markersize = 4, linewidth = 0.8, label = \"HP1_blood\")\n",
    "                    ax_BRCA1[patient_index].plot(x_hp2_blood, y_hp2_blood, \"-\", color = NORMAL_COLOR_HP2, markerfacecolor=NORMAL_COLOR_HP2, mec = NORMAL_COLOR_HP2, marker=\"^\", markersize = 4, linewidth = 0.8, label = \"HP2_blood\")\n",
    "            \n",
    "            \n",
    "            ax_BRCA1[patient_index].set_ylim(-0.05, 1.1)\n",
    "    \n",
    "            xlims = plt.gca().get_xlim()\n",
    "            \n",
    "            xticks = ax_BRCA1[patient_index].get_xticks()\n",
    "            xticks = np.linspace(*xlims, 6)\n",
    "            ax_BRCA1[patient_index].set_xticks(xticks)\n",
    "            \n",
    "\n",
    "            if patient_index >1:\n",
    "                \n",
    "                xticks = [str(int(value)) for value in xticks]\n",
    "                ax_BRCA1[patient_index].set_xticklabels(xticks)\n",
    "    \n",
    "                yticks = [0.0, 0.5, 1.0]\n",
    "                ax_BRCA1[patient_index].set_yticks(yticks)\n",
    "                yticks = [str(float(value)) for value in yticks]\n",
    "                ax_BRCA1[patient_index].set_yticklabels(yticks)\n",
    "    \n",
    "                ax_BRCA1[patient_index].tick_params(axis=\"both\")\n",
    "                ax_BRCA1[patient_index].set_xlabel(\"genomic coordinates\")\n",
    "                ax_BRCA1[patient_index].spines[\"right\"].set_visible(False) \n",
    "                ax_BRCA1[patient_index].spines[\"top\"].set_visible(False)\n",
    "                \n",
    "            else:\n",
    "                ax_BRCA1[patient_index].set_xticklabels(\"\")\n",
    "    \n",
    "                yticks = [0.0, 0.5, 1.0]\n",
    "                ax_BRCA1[patient_index].set_yticks(yticks)\n",
    "                yticks = [str(float(value)) for value in yticks]\n",
    "                ax_BRCA1[patient_index].set_yticklabels(yticks)\n",
    "    \n",
    "                ax_BRCA1[patient_index].tick_params(axis=\"both\")\n",
    "    \n",
    "            if patient_index == 1:\n",
    "                ax_BRCA1[patient_index].set_ylabel(\"methyl frequencies\")\n",
    "                \n",
    "            ax_BRCA1[patient_index].set_ylim(-0.1, 1.05)\n",
    "            ax_BRCA1[patient_index].spines[\"right\"].set_visible(False) \n",
    "            ax_BRCA1[patient_index].spines[\"top\"].set_visible(False)\n",
    "    \n",
    "            ax_BRCA1[patient_index].text(43126965, -0.05, patient, size = \"medium\", weight = \"bold\")\n",
    "\n",
    "        for i in range(3):\n",
    "            ax_BRCA1[i].set_xlim(ax_BRCA1[i].get_xlim()[::-1])\n",
    "    \n",
    "        plt.legend(frameon=False, loc=\"lower left\", bbox_to_anchor=(0.9, -0.05))\n",
    "        plt.tight_layout()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-03-05T01:01:50.688290Z",
     "iopub.status.busy": "2024-03-05T01:01:50.687438Z",
     "iopub.status.idle": "2024-03-05T01:01:50.702673Z",
     "shell.execute_reply": "2024-03-05T01:01:50.702314Z"
    }
   },
   "outputs": [],
   "source": [
    "###################\n",
    "##### RAD51C ######\n",
    "###################\n",
    "\n",
    "def generate_RAD51C_haplotype_plots(RAD51C_files, RAD51C_methylated_cases_BRC_OVA, RAD51C_methylated_cases_non_BRC_OVA, RAD51C_loh_cases):\n",
    "    \n",
    "    tumour_types = [\"BRC_OVA\", \"non_BRC_OVA\"]\n",
    "    \n",
    "    for tumour_type in tumour_types:\n",
    "        df_blood = pd.DataFrame()\n",
    "        df_tumour = pd.DataFrame()\n",
    "        \n",
    "        if tumour_type == \"BRC_OVA\":\n",
    "            ### Figure 5d\n",
    "            case_list = RAD51C_methylated_cases_BRC_OVA\n",
    "            \n",
    "        else:\n",
    "            ### Ex Fig 5c\n",
    "            case_list = RAD51C_methylated_cases_non_BRC_OVA\n",
    "        \n",
    "        files_RAD51C = RAD51C_files[tumour_type]\n",
    "        \n",
    "        if len(files_RAD51C[0]) >0:\n",
    "            df_blood = pd.read_csv(files_RAD51C[0], sep = \"\\t\", index_col = [\"gene\",\"start\", \"end\"])\n",
    "        if len(files_RAD51C[1])>0:\n",
    "            df_tumour = pd.read_csv(files_RAD51C[1], sep = \"\\t\", index_col = [\"gene\",\"start\", \"end\"])\n",
    "\n",
    "        logger.info(\"generating RAD51C haplotype plots for {}\".format(tumour_type))\n",
    "\n",
    "        fig_RAD51C, ax_RAD51C = plt.subplots(ncols=1, nrows=3,sharex = True, figsize = (8.5, 4.2), dpi = 600)\n",
    "    \n",
    "        for patient_index, patient in enumerate(case_list):\n",
    "            LOH = \"\"\n",
    "            hp1_column = f\"{patient}_HP1\"\n",
    "            hp2_column = f\"{patient}_HP2\"\n",
    "            \n",
    "            if patient in RAD51C_loh_cases:\n",
    "                LOH = \"yes\"\n",
    "            \n",
    "            \n",
    "            if not df_tumour.empty:\n",
    "        \n",
    "                hp1 = df_tumour[hp1_column]\n",
    "                hp2 = df_tumour[hp2_column]\n",
    "                xpos = [value[1] for value in df_tumour.index]\n",
    "        \n",
    "                ### LOESS SMOOTHING TUMOUR\n",
    "                filtered_hp1 = lowess(hp1, xpos, is_sorted=True, frac=0.25, it=0)\n",
    "                x_hp1 = filtered_hp1[:, 0]\n",
    "                y_hp1 = filtered_hp1[:, 1]\n",
    "        \n",
    "                filtered_hp2 = lowess(hp2, xpos, is_sorted=True, frac=0.25, it=0)\n",
    "                x_hp2 = filtered_hp2[:, 0]\n",
    "                y_hp2 = filtered_hp2[:, 1]\n",
    "    \n",
    "            if not df_blood.empty:\n",
    "                if hp1_column in df_blood.columns:\n",
    "                    hp1_blood = df_blood[hp1_column]\n",
    "                    hp2_blood = df_blood[hp2_column]\n",
    "                    xpos_blood = [value[1] for value in df_blood.index]\n",
    "        \n",
    "                    ### LOESS SMOOTHING BLOOD\n",
    "                    filtered_hp1_blood = lowess(hp1_blood, xpos_blood, is_sorted=True, frac=0.25, it=0)\n",
    "                    x_hp1_blood = filtered_hp1_blood[:, 0]\n",
    "                    y_hp1_blood = filtered_hp1_blood[:, 1]\n",
    "        \n",
    "                    filtered_hp2_blood = lowess(hp2_blood, xpos_blood, is_sorted=True, frac=0.25, it=0)\n",
    "                    x_hp2_blood = filtered_hp2_blood[:, 0]\n",
    "                    y_hp2_blood = filtered_hp2_blood[:, 1]\n",
    "                else:\n",
    "                    x_hp1_blood = []\n",
    "                    y_hp1_blood = []\n",
    "                    x_hp2_blood = []\n",
    "                    y_hp2_blood = []\n",
    "\n",
    "            if not df_tumour.empty:\n",
    "                if LOH == \"yes\":\n",
    "                    ax_RAD51C[patient_index].plot(x_hp1, y_hp1, \"-\", color = TUMOUR_COLOR_HP1, markerfacecolor=TUMOUR_COLOR_HP1, mec = TUMOUR_COLOR_HP1, marker=\"o\", markersize = 4, linewidth = 0.8, label = \"HP1_tumour\")\n",
    "                    ax_RAD51C[patient_index].plot(x_hp2, y_hp2, \"-\", color = TUMOUR_COLOR_HP2, markerfacecolor=\"none\", mec = TUMOUR_COLOR_HP2, mew = 0.5, marker=\"o\", markersize = 4.5, linewidth = 0.8, label = \"HP2_tumour\")\n",
    "                else:\n",
    "                    ax_RAD51C[patient_index].plot(x_hp1, y_hp1, \"-\", color = TUMOUR_COLOR_HP1, markerfacecolor=TUMOUR_COLOR_HP1, mec = TUMOUR_COLOR_HP1, marker=\"o\", markersize = 4, linewidth = 0.8, label = \"HP1_tumour\")\n",
    "                    ax_RAD51C[patient_index].plot(x_hp2, y_hp2, \"-\", color = TUMOUR_COLOR_HP2, markerfacecolor=TUMOUR_COLOR_HP2, mec = TUMOUR_COLOR_HP2, marker=\"o\", markersize = 4, linewidth = 0.8, label = \"HP2_tumour\")\n",
    "            if not df_blood.empty:\n",
    "                ax_RAD51C[patient_index].plot(x_hp1_blood, y_hp1_blood, \"-\", color = NORMAL_COLOR_HP1, markerfacecolor=NORMAL_COLOR_HP1, mec = NORMAL_COLOR_HP1, marker=\"^\", markersize = 4, linewidth = 0.8, label = \"HP1_blood\")\n",
    "                ax_RAD51C[patient_index].plot(x_hp2_blood, y_hp2_blood, \"-\", color = NORMAL_COLOR_HP2, markerfacecolor=NORMAL_COLOR_HP2, mec = NORMAL_COLOR_HP2, marker=\"^\", markersize = 4, linewidth = 0.8, label = \"HP2_blood\")\n",
    "                \n",
    "            ax_RAD51C[patient_index].set_ylim(-0.05, 1.2)\n",
    "    \n",
    "            xlims = plt.gca().get_xlim()\n",
    "            xticks = ax_RAD51C[patient_index].get_xticks()\n",
    "            xticks = np.linspace(*xlims, 6)\n",
    "            ax_RAD51C[patient_index].set_xticks(xticks)\n",
    "\n",
    "            if patient_index >1:\n",
    "\n",
    "                xticks = [str(int(value)) for value in xticks]\n",
    "                ax_RAD51C[patient_index].set_xticklabels(xticks)\n",
    "    \n",
    "                yticks = [0.0, 0.5, 1.0]\n",
    "                ax_RAD51C[patient_index].set_yticks(yticks)\n",
    "                yticks = [str(float(value)) for value in yticks]\n",
    "                ax_RAD51C[patient_index].set_yticklabels(yticks)\n",
    "    \n",
    "                ax_RAD51C[patient_index].tick_params(axis=\"both\")\n",
    "                ax_RAD51C[patient_index].set_xlabel(\"genomic coordinates\")\n",
    "                ax_RAD51C[patient_index].spines[\"right\"].set_visible(False) \n",
    "                ax_RAD51C[patient_index].spines[\"top\"].set_visible(False)\n",
    "            else:\n",
    "                ax_RAD51C[patient_index].set_xticklabels(\"\")\n",
    "    \n",
    "                yticks = [0.0, 0.5, 1.0]\n",
    "                ax_RAD51C[patient_index].set_yticks(yticks)\n",
    "                yticks = [str(float(value)) for value in yticks]\n",
    "                ax_RAD51C[patient_index].set_yticklabels(yticks)\n",
    "    \n",
    "                ax_RAD51C[patient_index].tick_params(axis=\"both\")\n",
    "            \n",
    "    \n",
    "            if patient_index == 1:\n",
    "                ax_RAD51C[patient_index].set_ylabel(\"methyl frequencies\")\n",
    "                # ax[patient_index].legend(bbox_to_anchor=(1.04, 0.5), loc = \"center left\", fontsize = 9)\n",
    "                \n",
    "            ax_RAD51C[patient_index].set_ylim(-0.1, 1.05)\n",
    "            ax_RAD51C[patient_index].spines[\"right\"].set_visible(False) \n",
    "            ax_RAD51C[patient_index].spines[\"top\"].set_visible(False)\n",
    "    \n",
    "            ax_RAD51C[patient_index].text(58690993, -0.05, patient, size = \"medium\", weight = \"bold\")\n",
    "    \n",
    "        plt.legend(frameon=False, loc=\"lower left\", bbox_to_anchor=(0.9, -0.05))\n",
    "        plt.tight_layout()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-03-05T01:01:50.705015Z",
     "iopub.status.busy": "2024-03-05T01:01:50.704705Z",
     "iopub.status.idle": "2024-03-05T01:01:50.716307Z",
     "shell.execute_reply": "2024-03-05T01:01:50.715945Z"
    }
   },
   "outputs": [],
   "source": [
    "##################\n",
    "##### MLH1 #######\n",
    "##################\n",
    "\n",
    "def generate_MLH1_haplotype_plots(filename_MLH1_blood, filename_MLH1_tumour, MLH1_methylated_cases, MLH1_loh_case):\n",
    "\n",
    "    logger.info(\"generating MLH1 haplotype plots\")\n",
    "\n",
    "    df_blood = pd.read_csv(filename_MLH1_blood, sep = \"\\t\", index_col = [\"gene\",\"start\", \"end\"])\n",
    "    df_tumour = pd.read_csv(filename_MLH1_tumour, sep = \"\\t\", index_col = [\"gene\",\"start\", \"end\"])\n",
    "\n",
    "    fig_MLH1, ax_MLH1 = plt.subplots(ncols=1, nrows=2, sharex = True, figsize = (5.2, 4), dpi = 600)\n",
    "    \n",
    "    for patient_index, patient in enumerate(MLH1_methylated_cases):\n",
    "        LOH = \"\"\n",
    "        if patient in MLH1_loh_case:\n",
    "            LOH = \"yes\"\n",
    "        \n",
    "        if not df_tumour.empty:\n",
    "            hp1_column = f\"{patient}_HP1\"\n",
    "            hp2_column = f\"{patient}_HP2\"\n",
    "        \n",
    "            hp1 = df_tumour[hp1_column]\n",
    "            hp2 = df_tumour[hp2_column]\n",
    "            xpos = [value[1] for value in df_tumour.index]\n",
    "        \n",
    "            filtered_hp1 = lowess(hp1, xpos, is_sorted=True, frac=0.25, it=0)\n",
    "            x_hp1 = filtered_hp1[:, 0]\n",
    "            y_hp1 = filtered_hp1[:, 1]\n",
    "        \n",
    "            filtered_hp2 = lowess(hp2, xpos, is_sorted=True, frac=0.25, it=0)\n",
    "            x_hp2 = filtered_hp2[:, 0]\n",
    "            y_hp2 = filtered_hp2[:, 1]\n",
    "\n",
    "\n",
    "        if not df_blood.empty:\n",
    "            if hp1_column in df_blood.columns:\n",
    "                hp1_blood = df_blood[hp1_column]\n",
    "                hp2_blood = df_blood[hp2_column]\n",
    "                xpos_blood = [value[1] for value in df_blood.index]\n",
    "        \n",
    "                ### LOESS SMOOTHING BLOOD\n",
    "                filtered_hp1_blood = lowess(hp1_blood, xpos_blood, is_sorted=True, frac=0.25, it=0)\n",
    "                x_hp1_blood = filtered_hp1_blood[:, 0]\n",
    "                y_hp1_blood = filtered_hp1_blood[:, 1]\n",
    "            \n",
    "                filtered_hp2_blood = lowess(hp2_blood, xpos_blood, is_sorted=True, frac=0.25, it=0)\n",
    "                x_hp2_blood = filtered_hp2_blood[:, 0]\n",
    "                y_hp2_blood = filtered_hp2_blood[:, 1]\n",
    "            else:\n",
    "                x_hp1_blood = []\n",
    "                y_hp1_blood = []\n",
    "                x_hp2_blood = []\n",
    "                y_hp2_blood = []\n",
    "\n",
    "        if LOH == \"yes\":\n",
    "\n",
    "            if not df_tumour.empty:\n",
    "                ax_MLH1[patient_index].plot(x_hp1, y_hp1, \"-\", color = TUMOUR_COLOR_HP1, markerfacecolor=\"none\", mec = TUMOUR_COLOR_HP1, mew = 0.5, marker=\"o\", markersize = 4.5, linewidth = 0.8, label = \"HP1_tumour\")\n",
    "                ax_MLH1[patient_index].plot(x_hp2, y_hp2, \"-\", color = TUMOUR_COLOR_HP2, markerfacecolor=TUMOUR_COLOR_HP2, mec = TUMOUR_COLOR_HP2, marker=\"o\", markersize = 4, linewidth = 0.8, label = \"HP2_tumour\")\n",
    "            \n",
    "            if not df_blood.empty:\n",
    "                ax_MLH1[patient_index].plot(x_hp2_blood, y_hp2_blood, \"-\", color = NORMAL_COLOR_HP1, markerfacecolor=NORMAL_COLOR_HP1, mec = NORMAL_COLOR_HP1, marker=\"^\", markersize = 4, linewidth = 0.8, label = \"HP1_blood\")\n",
    "                ax_MLH1[patient_index].plot(x_hp1_blood, y_hp1_blood, \"-\", color = NORMAL_COLOR_HP2, markerfacecolor=NORMAL_COLOR_HP2, mec = NORMAL_COLOR_HP2, marker=\"^\", markersize = 4, linewidth = 0.8, label = \"HP2_blood\")\n",
    "            \n",
    "        else:\n",
    "            if not df_tumour.empty:\n",
    "                ax_MLH1[patient_index].plot(x_hp1, y_hp1, \"-\", color = TUMOUR_COLOR_HP1, markerfacecolor=TUMOUR_COLOR_HP1, mec = TUMOUR_COLOR_HP1, marker=\"o\", markersize = 4, linewidth = 0.8, label = \"HP1_tumour\")\n",
    "                ax_MLH1[patient_index].plot(x_hp2, y_hp2, \"-\", color = TUMOUR_COLOR_HP2, markerfacecolor=TUMOUR_COLOR_HP2, mec = TUMOUR_COLOR_HP2, marker=\"o\", markersize = 4, linewidth = 0.8, label = \"HP2_tumour\")\n",
    "\n",
    "            if not df_blood.empty:\n",
    "                ax_MLH1[patient_index].plot(x_hp2_blood, y_hp2_blood, \"-\", color = NORMAL_COLOR_HP1, markerfacecolor=NORMAL_COLOR_HP1, mec = NORMAL_COLOR_HP1, marker=\"^\", markersize = 4, linewidth = 0.8, label = \"HP1_blood\")\n",
    "                ax_MLH1[patient_index].plot(x_hp1_blood, y_hp1_blood, \"-\", color = NORMAL_COLOR_HP2, markerfacecolor=NORMAL_COLOR_HP2, mec = NORMAL_COLOR_HP2, marker=\"^\", markersize = 4, linewidth = 0.8, label = \"HP2_blood\")\n",
    "            \n",
    "        ax_MLH1[patient_index].set_ylim(-0.05, 1.1)\n",
    "\n",
    "        ax_MLH1[patient_index].set_ylabel(\"methyl frequencies\")\n",
    "\n",
    "        xlims = plt.gca().get_xlim()\n",
    "        xticks = ax_MLH1[patient_index].get_xticks()\n",
    "        xticks = np.linspace(*xlims, 5)\n",
    "        ax_MLH1[patient_index].set_xticks(xticks)\n",
    "\n",
    "        if patient_index >0:\n",
    "\n",
    "            xticks = [str(int(value)) for value in xticks]\n",
    "            ax_MLH1[patient_index].set_xticklabels(xticks)\n",
    "            ax_MLH1[patient_index].tick_params(axis=\"both\")\n",
    "            ax_MLH1[patient_index].set_xlabel(\"genomic coordinates\")\n",
    "            ax_MLH1[patient_index].spines[\"right\"].set_visible(False) \n",
    "            ax_MLH1[patient_index].spines[\"top\"].set_visible(False)\n",
    "        else:\n",
    "            ax_MLH1[patient_index].set_xticklabels(\"\")\n",
    "            ax_MLH1[patient_index].tick_params(axis=\"both\")\n",
    "            \n",
    "        ax_MLH1[patient_index].spines[\"right\"].set_visible(False) \n",
    "        ax_MLH1[patient_index].spines[\"top\"].set_visible(False)\n",
    "        ax_MLH1[patient_index].text(36991870, -0.03, patient, size = \"medium\", weight = \"bold\")\n",
    "\n",
    "    plt.legend(frameon=False, loc=\"lower left\", bbox_to_anchor=(0.75, -0.05))\n",
    "    plt.tight_layout()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-03-05T01:01:50.718604Z",
     "iopub.status.busy": "2024-03-05T01:01:50.718280Z",
     "iopub.status.idle": "2024-03-05T01:01:50.943668Z",
     "shell.execute_reply": "2024-03-05T01:01:50.943214Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:__main__:generating BRCA1 haplotype plots for BRC_OVA\n",
      "CRITICAL:__main__:Failure: <urlopen error [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1006)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.11/urllib/request.py\", line 1348, in do_open\n",
      "    h.request(req.get_method(), req.selector, req.data, headers,\n",
      "  File \"/opt/conda/lib/python3.11/http/client.py\", line 1286, in request\n",
      "    self._send_request(method, url, body, headers, encode_chunked)\n",
      "  File \"/opt/conda/lib/python3.11/http/client.py\", line 1332, in _send_request\n",
      "    self.endheaders(body, encode_chunked=encode_chunked)\n",
      "  File \"/opt/conda/lib/python3.11/http/client.py\", line 1281, in endheaders\n",
      "    self._send_output(message_body, encode_chunked=encode_chunked)\n",
      "  File \"/opt/conda/lib/python3.11/http/client.py\", line 1041, in _send_output\n",
      "    self.send(msg)\n",
      "  File \"/opt/conda/lib/python3.11/http/client.py\", line 979, in send\n",
      "    self.connect()\n",
      "  File \"/opt/conda/lib/python3.11/http/client.py\", line 1458, in connect\n",
      "    self.sock = self._context.wrap_socket(self.sock,\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/opt/conda/lib/python3.11/ssl.py\", line 517, in wrap_socket\n",
      "    return self.sslsocket_class._create(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/opt/conda/lib/python3.11/ssl.py\", line 1108, in _create\n",
      "    self.do_handshake()\n",
      "  File \"/opt/conda/lib/python3.11/ssl.py\", line 1379, in do_handshake\n",
      "    self._sslobj.do_handshake()\n",
      "ssl.SSLError: [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1006)\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/tmp/ipykernel_48/2671134438.py\", line 37, in <module>\n",
      "    main()\n",
      "  File \"/tmp/ipykernel_48/2671134438.py\", line 14, in main\n",
      "    generate_BRCA1_haplotype_plots(BRCA1_files, BRCA1_METHYLATED_CASES_BRC_OVA, BRCA1_METHYLATED_CASES_NON_BRC_OVA, BRCA1_LOH_CASES)\n",
      "  File \"/tmp/ipykernel_48/1885639513.py\", line 25, in generate_BRCA1_haplotype_plots\n",
      "    df_blood = pd.read_csv(files_BRCA1[0], sep = \"\\t\", index_col = [\"gene\",\"start\", \"end\"])\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/opt/conda/lib/python3.11/site-packages/pandas/io/parsers/readers.py\", line 1026, in read_csv\n",
      "    return _read(filepath_or_buffer, kwds)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/opt/conda/lib/python3.11/site-packages/pandas/io/parsers/readers.py\", line 620, in _read\n",
      "    parser = TextFileReader(filepath_or_buffer, **kwds)\n",
      "             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/opt/conda/lib/python3.11/site-packages/pandas/io/parsers/readers.py\", line 1620, in __init__\n",
      "    self._engine = self._make_engine(f, self.engine)\n",
      "                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/opt/conda/lib/python3.11/site-packages/pandas/io/parsers/readers.py\", line 1880, in _make_engine\n",
      "    self.handles = get_handle(\n",
      "                   ^^^^^^^^^^^\n",
      "  File \"/opt/conda/lib/python3.11/site-packages/pandas/io/common.py\", line 728, in get_handle\n",
      "    ioargs = _get_filepath_or_buffer(\n",
      "             ^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/opt/conda/lib/python3.11/site-packages/pandas/io/common.py\", line 384, in _get_filepath_or_buffer\n",
      "    with urlopen(req_info) as req:\n",
      "         ^^^^^^^^^^^^^^^^^\n",
      "  File \"/opt/conda/lib/python3.11/site-packages/pandas/io/common.py\", line 289, in urlopen\n",
      "    return urllib.request.urlopen(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/opt/conda/lib/python3.11/urllib/request.py\", line 216, in urlopen\n",
      "    return opener.open(url, data, timeout)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/opt/conda/lib/python3.11/urllib/request.py\", line 519, in open\n",
      "    response = self._open(req, data)\n",
      "               ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/opt/conda/lib/python3.11/urllib/request.py\", line 536, in _open\n",
      "    result = self._call_chain(self.handle_open, protocol, protocol +\n",
      "             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/opt/conda/lib/python3.11/urllib/request.py\", line 496, in _call_chain\n",
      "    result = func(*args)\n",
      "             ^^^^^^^^^^^\n",
      "  File \"/opt/conda/lib/python3.11/urllib/request.py\", line 1391, in https_open\n",
      "    return self.do_open(http.client.HTTPSConnection, req,\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/opt/conda/lib/python3.11/urllib/request.py\", line 1351, in do_open\n",
      "    raise URLError(err)\n",
      "urllib.error.URLError: <urlopen error [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1006)>\n"
     ]
    }
   ],
   "source": [
    "def main():\n",
    "    \n",
    "    ### BRCA1 Fig 5c\n",
    "    filename_BRCA1_BRC_OVA_blood = r\"https://www.bcgsc.ca/downloads/nanopore_pog/ts_methylation/BRCA1_blood_haplotype_methyl_freq_BRC_OVA.tsv\"\n",
    "    filename_BRCA1_BRC_OVA_tumour = r\"https://www.bcgsc.ca/downloads/nanopore_pog/ts_methylation/BRCA1_tumour_haplotype_methyl_freq_BRC_OVA.tsv\"\n",
    "    ### BRCA1 Ex Fig5b\n",
    "    filename_BRCA1_non_BRC_OVA_blood = r\"\"\n",
    "    filename_BRCA1_non_BRC_OVA_tumour = r\"https://www.bcgsc.ca/downloads/nanopore_pog/ts_methylation//BRCA1_tumour_haplotype_methyl_freq_non_BRC_OVA.tsv\"\n",
    "\n",
    "    BRCA1_files = {\"BRC_OVA\": [filename_BRCA1_BRC_OVA_blood, filename_BRCA1_BRC_OVA_tumour], \n",
    "                  \"non_BRC_OVA\":[filename_BRCA1_non_BRC_OVA_blood, filename_BRCA1_non_BRC_OVA_tumour]}\n",
    "\n",
    "\n",
    "    generate_BRCA1_haplotype_plots(BRCA1_files, BRCA1_METHYLATED_CASES_BRC_OVA, BRCA1_METHYLATED_CASES_NON_BRC_OVA, BRCA1_LOH_CASES)\n",
    "\n",
    "    ### RAD51C Fig 5d\n",
    "    filename_RAD51C_BRC_OVA_blood = r\"https://www.bcgsc.ca/downloads/nanopore_pog/ts_methylation/RAD51C_blood_haplotype_methyl_freq_BRC_OVA.tsv\"\n",
    "    filename_RAD51C_BRC_OVA_tumour = r\"https://www.bcgsc.ca/downloads/nanopore_pog/ts_methylation/RAD51C_tumour_haplotype_methyl_freq_BRC_OVA.tsv\"\n",
    "    ### RAD51C Ex Fig5c\n",
    "    filename_RAD51C_non_BRC_OVA_blood = r\"https://www.bcgsc.ca/downloads/nanopore_pog/ts_methylation/RAD51C_blood_haplotype_methyl_freq_non_BRC_OVA.tsv\"\n",
    "    filename_RAD51C_non_BRC_OVA_tumour = r\"https://www.bcgsc.ca/downloads/nanopore_pog/ts_methylation/RAD51C_tumour_haplotype_methyl_freq_non_BRC_OVA.tsv\"\n",
    "   \n",
    "    RAD51C_files = {\"BRC_OVA\": [filename_RAD51C_BRC_OVA_blood, filename_RAD51C_BRC_OVA_tumour], \n",
    "                  \"non_BRC_OVA\":[filename_RAD51C_non_BRC_OVA_blood, filename_RAD51C_non_BRC_OVA_tumour]}\n",
    "\n",
    "    \n",
    "    generate_RAD51C_haplotype_plots(RAD51C_files, RAD51C_METHYLATED_CASES_BRC_OVA, RAD51C_METHYLATED_CASES_NON_BRC_OVA, RAD51C_LOH_CASES)\n",
    "\n",
    "    ### MLH1 Fig 5f\n",
    "    filename_MLH1_blood = r\"https://www.bcgsc.ca/downloads/nanopore_pog/ts_methylation/MLH1_blood_haplotype_methyl_freq.tsv\"\n",
    "    filename_MLH1_tumour = r\"https://www.bcgsc.ca/downloads/nanopore_pog/ts_methylation/MLH1_tumour_haplotype_methyl_freq.tsv\"\n",
    "\n",
    "    generate_MLH1_haplotype_plots(filename_MLH1_blood, filename_MLH1_tumour, MLH1_METHYLATED_CASES, MLH1_LOH_CASE)\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    try:\n",
    "        main()\n",
    "    except Exception as err:\n",
    "        logger.critical(\"Failure: %s\", err, exc_info=True)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
